amplipi/streams/internet_radio.py

```diff
@@ -137,9 +139,15 @@ def send_cmd(self, cmd):
       pass
 
   def validate_stream(self, **kwargs):
-    URL_LIKE = r'^https?://[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)$'
-    if 'url' in kwargs and not re.fullmatch(URL_LIKE, kwargs['url']):
-      raise InvalidStreamField("url", "invalid url")
+    if 'url' in kwargs and kwargs['url']:
+      if not validators.url(kwargs['url']):
+        raise InvalidStreamField("url", "invalid url")
+      if urlparse(kwargs['url']).scheme not in ['http', 'https']:
+        raise InvalidStreamField("url", "unsupported protocol/scheme in url")
+
     # Logo is Optional[str]
-    if 'logo' in kwargs and kwargs['logo'] and not re.fullmatch(URL_LIKE, kwargs['logo']):
-      raise InvalidStreamField("logo", "invalid logo url")
+    if 'logo' in kwargs and kwargs['logo']:
+      if not validators.url(kwargs['logo']):
```

Comment:
this interface changes sometime after 0.20 - `url()` will instead raise an exception. We'll at least catch it when we upgrade ezpz thanks to testing, but just to note cuz this being way different while staying within the major version (even tho it's 0) confused me for way longer than it should have.